### PR TITLE
remove ESH documentation

### PR DESCRIPTION
**Pull request by kaikreuzer**
This will rather have to be migrated to openhab-docs

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by maggu2810** [comment]
How do you intent to differ between framework and product documentation?
How should I know if a documentation is related to the framework or not?

**Comment by kaikreuzer** [comment]
Well, there is enough empty space on https://www.openhab.org/docs/developer/ to explain this :-).

And so far we didn't do a proper split at ESH either - most documentation was on how to develop bindings and what the framework provides. But (correct me if I am wrong): We had hardly any docs on how to do coding for the framework itself.

**Comment by maggu2810** [comment]
The documentation contains enough stuff (IMHO) to explain the current code base. If you think that the documentation has been "not good enough", we should perhaps try to improve the documentation and not solve it by moving it.

I am using the framework for different products for different companies.
I have to create custom branches in forks of the upstream repo, do further testing, add fixes, apply custom changes, cherry-pick some features, ...
After that a release will be build and the sources will be tagged.
If the client or other developers needs to develop custom stuff I can point to the tag that contains the sources and the documentation that fits to the correct version.

How can this be done if the documentation resides in another repository?
Why should the documentation of a framework be mixed with the documentation of a product and placed in a separate repository?
How can I provide my clients a documentation of the framework without removing all the non framework documentation again?
If I cherry pick a added feature previously also the added documentation has been part of the back port as it has been provided with the code changes.

As we agreed to keep the framework separated, I am really not happy if the documentation of the framework is removed and mixed up with the other documentation.
I understand the intention to keep all documentation in one repository if we see the whole openHAB repositories as "one thing" that is all versioned together etc.
But if someone wants to use just the framework and would like that its forks contains also the documentation of the code base I don't think this is possible.

**Comment by kaikreuzer** [comment]
> we should perhaps try to improve the documentation and not solve it by moving it.

We should not keep stuff that does not belong here, but only add things that specifically concern the framework and which is already up to date with the code base.
What's currently under docs is all infrastructure to build the https://www.eclipse.org/smarthome/documentation/ website. It was never intended to be referenced directly. 
The equivalent of https://www.eclipse.org/smarthome/documentation/ is https://www.openhab.org/docs/developer/. This page can also be sourced from markdown files in the repo, so it isn't said, where the documentation source resides.

> How can this be done if the documentation resides in another repository?

We keep the docs versioned along with the releases, see e.g.

screenshot 2019-01-24 at 09 33 11

> Why should the documentation of a framework be mixed with the documentation of a product

Because "openHAB Development" means development of many different things (core, addons, cloud, Android, Alexa skills, etc.) - there should be one place (which is the developer guide) that helps finding the desired information. Again, I am not saying what's the best place for the doc sources.

> As we agreed to keep the framework separated

We agreed to keep openhab-core in a "framework-style" to be usable independently of the rest of openHAB. But we also agreed that we want to simplify the overall openHAB infrastructure by not running ESH as a separate project anymore (with its own versioning, website, releases, etc.). So openhab-core IS a part of the openHAB project.

TL;DR: I definitely think the current docs folder must be removed before the merge. I am not against adding documentation to the openhab-core repo, but this should be up-to-date and it should be clarified if and how it is published to https://www.openhab.org/docs/developer - something that should be discussed with @Confectrician at least.

**Comment by maggu2810** [comment]
> We should not keep stuff that does not belong here, but only add things that specifically concern the framework and which is already up to date with the code base.
> What's currently under docs is all infrastructure to build the https://www.eclipse.org/smarthome/documentation/ website. It was never intended to be referenced directly.
> The equivalent of https://www.eclipse.org/smarthome/documentation/ is https://www.openhab.org/docs/developer/. 

You did not remove the only the part to build the documentation, you remove also the documentation itself, so all files that resides in the `documentation` folder of ´docs` (e.g. the xml-reference.md file etc.).

> This page can also be sourced from markdown files in the repo, so it isn't said, where the documentation source resides.

So, why removing the markdown files?

> We keep the docs versioned along with the releases, see e.g.

I am pretty sure you know that a commercial product needs other releases then the official ones.

> Because "openHAB Development" means development of many different things (core, addons, cloud, Android, Alexa skills, etc.) - there should be one place (which is the developer guide) that helps finding the desired information. Again, I am not saying what's the best place for the doc sources.

My concerns are not about the place the generated stuff is hosted or found.
I agree that the build result should all be placed in one place.
But I need to repeat, you removed the doc sources with that commit.

>  I definitely think the current docs folder must be removed before the merge. I am not against adding documentation to the openhab-core repo, but this should be up-to-date

The documentation and the sources of the ESH projects has been migrated.
So the documentation of ESH is not up to date to the sources of ESH?
Or do you point the the occurrences of "Eclipse SmartHome" in the documentation?
Then we should take the time to fix it similar to the other changes you want to be done before an initial merge.

"We remove it now and add it later in a up-to-date version"...
I fear this will never happen if we start removing it.

> We agreed to keep openhab-core in a "framework-style" to be usable independently of the rest of openHAB. 

IMHO this is still another argument for keeping the md files in that repository.
The documentation of sources of the framework should be placed into one repo, so if I fork, cherry pick, ... the repository the framework documentation is part of it.

**Comment by kaikreuzer** [comment]
> So, why removing the markdown files?

Because they were meant as a source for the ESH website (e.g. using html instead of md references) many of them are out-of-date, and the structure most likely needs to be reconsidered for openhab-core. I simply cannot spend the time to create new documentation structure for openhab-core right now and I don't think it is urgent as all is still available at ESH.
All I want to do is to make sure that no useless/outdated/invalid content is added to openhab-core through your PR in the first place.

> So the documentation of ESH is not up to date to the sources of ESH?

As I said: A lot is about developing bindings, not about the framework.

> But I need to repeat, you removed the doc sources with that commit.

No, I just do not want them to be added as-is to openhab-core with this PR. They are still identically available at https://github.com/eclipse/smarthome/tree/master/docs/documentation, so there's no real sense in duplicating it (without having it refactored first).

> I fear this will never happen if we start removing it.

The topic is already scheduled, see https://community.openhab.org/t/migration-due-to-esh-reintegration/64754.

I fear we will never merge, if we first want to wait for the docs to be updated.
